src/Integration/LocalServices/ProjectSystemHelper.cs

```diff
@@ -56,24 +56,19 @@ internal class ProjectSystemHelper : IProjectSystemHelper
         internal const uint SolutionItemResourceId = 13450;
 
         private readonly IServiceProvider serviceProvider;
+        private readonly IVsSolution solution;
 
         public ProjectSystemHelper(IServiceProvider serviceProvider)
         {
-            if (serviceProvider == null)
-            {
-                throw new ArgumentNullException(nameof(serviceProvider));
-            }
+            this.serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
 
-            this.serviceProvider = serviceProvider;
+            solution = this.serviceProvider.GetService<SVsSolution, IVsSolution>();
```

Comment:
I decided to fetch this in the ctor, so `EnumerateProjects` wouldn't have to get parameters